### ISSUE TITLE
Profile explorer tile code

### ISSUE THREAD

**Issue by martin-ueding** [martin-ueding/geo-activity-playground#48]
When opening the explorer tiles at zoom 17, it takes pretty long. Use py-spy to find out where it takes so long.

**Comment by martin-ueding** [comment]
I've used py-spy and speedscope to find out what the problem is. And apparently it takes super long to figure out the tile history:

![Screenshot_20231207_200625](https://github.com/martin-ueding/geo-activity-playground/assets/976924/9d502af8-6779-4379-9f10-bd068124c7d7)

Then I've changed this from a Pandas data frame with linear access time to a hashmap with constant access time. And then it is like 43 times faster:

![Screenshot_20231207_200508](https://github.com/martin-ueding/geo-activity-playground/assets/976924/23edf54a-620d-4add-a30e-b5956ed16c72)

Emitting the GeoJSON just takes some time. Perhaps there is a bit one can do. Also perhaps cache some other things.
